src/xwayland/XWM.cpp

```diff
@@ -891,6 +891,9 @@ void CXWM::activateSurface(SP<CXWaylandSurface> surf, bool activate) {
 }
 
 void CXWM::sendState(SP<CXWaylandSurface> surf) {
+    if (surf->fullscreen)
+        surf->setWithdrawn(surf != focusedSurface);
```

Comment:
this is not correct. Generally, we don't want to say a surface is withdrawn as long as its mapped on our side, ever.

Withdrawn is like "minimized to tray", from what I understand.

IMO we should just always set it to `false`, as long as the surface is mapped, if that doesn't break anything.

Reply:
This is what happens with proton games in i3. And that is what makes the issue go away. Maybe the withdrawn state should come from some unhandled event but I don't know which one it is.

Reply:
Whether or not it should always be set to false, the current patch does fix the black screen issue in all the games I've experienced it in.

Reply:
The question is whether changing `surf != focusedSurface` to `!surf->mapped` doesn't fix the issue as well, while being more of what we want.

Reply:
It turned out that resending normal state is enough. Still might not be 100% correct but works. Also added some debug logging.